### PR TITLE
Setting delete at

### PR DESCRIPTION
**Pull request by bdunne**
- Switch from fog to rest-client
- Add the "X-Delete-At" header to nightly uploads so that images are automatically expired and cleaned up after 8 weeks

### CODE REVIEW

**Comment by bdunne** [comment]
Done

**Comment by Fryguy** [comment]
Was fog not in the Gemfile?  I expected a deletion of fog from there.

File: scripts/uploader.rb [hunk 1]
```diff
@@ -1,46 +1,102 @@
...
       appliances.each do |appliance|
         # Skip badly named appliances, missing git sha: manageiq-openstack-master-201407142000-.qc2
-        next unless appliance.match(/.+-[0-9]{12}-[0-9a-fA-F]+/)
-        key = uploaded_filename(appliance)
-        puts "Uploading #{appliance} as #{key}..."
-        upload_file = File.expand_path(appliance)
-        file = container.files.create :key => key, :body => File.open(upload_file, "r")
-        url = file.public_url if file
-        puts "Uploading #{appliance} as #{key}...complete: #{url}"
+        unless appliance.match(/.+-[0-9]{12}-[0-9a-fA-F]+/)
```
Comment (Fryguy) [inline scripts/uploader.rb]: Minor, but since we don't actually use any of the match contents, better to use `.match?`

File: scripts/uploader.rb [hunk 2]
```diff
@@ -1,46 +1,102 @@
...
+
+        puts "Uploading #{appliance} as #{destination_name}..."
+
+        destination_url = url(destination_name)
+        source_hash = Digest::MD5.file(source_name).hexdigest
+
+        upload_headers = headers.merge("ETag" => source_hash)
+        unless release?
+          image_date = destination_name.split("-")[-2]
+          delete_at  = (DateTime.parse(image_date) + 8.weeks)
```
Comment (Fryguy) [inline scripts/uploader.rb]: Minor but can we make the `8.weeks` a constant?  That way if we want to change it, it's clearer where to find it.

File: scripts/uploader.rb [hunk 3]
```diff
@@ -1,46 +1,102 @@
...
+
+    def release?
+      type == "release"
+    end
+
+    def config
+      @config ||= YAML.load_file("#{__dir__}/../config/upload.yml")
+    end
+
+    def login
```
Comment (Fryguy) [inline scripts/uploader.rb]: Where do we call `login`?  Is it just a side affect of calling headers?  If so, that feels kind of weird as opposed to being explicit about it.

**Comment by bdunne** [comment]
> Was fog not in the Gemfile? I expected a deletion of fog from there.

Nope, but [removed it from here](https://github.com/ManageIQ/manageiq-appliance-build/pull/352/files#diff-04c6e90faac2675aa89e2176d2eec7d8L269)

**Comment by simaishi** [comment]
We're building ivanchuk nightly as well, so adding `ivanchuk/yes`.

**Comment by simaishi** [comment]
Ivanchuk backport details:

```
$ git log -1
commit 51bbe9e43d9cc8a0c7e26b71637e2c804a6e77ec
Author: Jason Frey <jfrey@redhat.com>
Date:   Tue Oct 1 16:07:22 2019 -0400

    Merge pull request #352 from bdunne/setting_delete_at
    
    Setting delete at
    
    (cherry picked from commit bea14ca25e23740f951dc0bb7353508a7a4218f3)
```

**Pull request merged**